developer-handbook/github-process/pull-request-process.md

```diff
@@ -0,0 +1,36 @@
+# Pull Request creation
+
+Here are recommendations to consider from the moment you create a new branch to work on.
+
+## Pre-creation considerations
+
+**Consider creating your Pull Request when you're 30-50% through**
+_If it's a big piece_, consider creating your PR when you're 30-50% through. Create it as a draft, so no one gets notified of every commit. 
+
+This will let others gain some familiarity with the code changes, and might allow them to flag things in advance. 
+
+This also ensures that the lottery factor isn't a problem; if you can't be the one to finish the work, others have eyes on it.
+
+**Pull Requests should be small as possible**
+If you're working on something big, consider splitting it in multiple PRs. If that's not possible, aim to split your commits logically, with meaningful commit messages.
+
+**Address `TODOs`**
+If there are `TODO`s in the changed files, even if they were there before you started your work: it's your responsability to check they have a chance to get addressed. 
+
+If they aren't as urgent or related to the original ticket, they don't need to be addressed in your PR (and probably shouldn't be). But every `TODO` should have a Github issue linked, so create one and add its link as a comment if it's not the case. Ensure it's on the team's radar if it's a bigger piece of work. 
```

Comment:
Consider whether the TODO belongs in a PR comment, rather than a code comment.

i.e. are you writing the TODO so that a reviewer understands that you know something should be different but is not necessarily pertinent to the task at hand?

Reply:
To double-check this point, if a TODO belongs to the PR, do we mean that the PR shouldn't get merged until it gets addressed?

Reply:
I think "addressed" in this context means "fixed or a separate ticket has been written".  Whether leaving it for a separate ticket/PR is acceptable can be up to the reviewers to agree/disagree.

The sentiment expressed by raising a (non-draft) PR is that the author thinks it's good to go, so any comments by the author should mean "This might not be right, but I think it's good enough".  However, as part of self-review, one might put comments in a draft PR that do need to be fixed before merging, and might be overlooked when pressing the "Ready for Review" button.

The sentiment expressed in a TODO is "I know this is imperfect/incomplete, but I haven't done anything about it yet".  I think it's better to state this in a review comment and linking it to a separate ticket, with the meaning "I'm not going to fix it in this PR, here's the ticket to fix it later", rather than in a code comment.

The problem with TODOs and ticket links in code comments is that they get stale in weird ways, whereas if they are in PR comments, we can always see the actual context.

Reply:
I understand what you mean, we definitely have 3year old+ TODOs in the code. One of the "pros" was that they were findeable that way, whereas a comment in a PR that is 3 years old shall probably never be seen again. The important thing is then just to ensure a ticket is created and in a backlog somewhere.

In-code TODOs also add context to a piece of code I might see 6 months later that I find a bit wonky, and give me permission to refactor?

But I think you're saying both scenarios _are_ valid, I'm just unsure how to phrase it. Could you help me?

Reply:
> The important thing is then just to ensure a ticket is created and in a backlog somewhere.

Agree with this, over adding TODO comments to code. My feeling is that we should avoid adding any "TODO" at all to `main`, instead:

- Create a ticket to do the work that can be prioritised
- Fix the issue before merging
- Add a comment describing the peculiarity or optimisation for context for the next person to read without implying the next person should fix it

In my opinion the issue is a TODO adds a "here is a bit of work you should do later" without any sense of priority or severity of the problem it indicates.  Adding an issue to the backlog allows the work to be properly prioritised in view of delivery/product folk.

Reply:
I do like that.... If the others are good to go for that I'd put that in.

Reply:
(amendment to prior comment😄 ) I do like that, very action focussed. If no one is objecting to this, I'll put that in.

Reply:
Now amended, section changed completely.